### PR TITLE
feat: expose headerNameToString

### PR DESCRIPTION
**Pull request by tsctx**
This is useful when processing header name.

### CODE REVIEW

File: types/util.d.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    headers: H,
+    obj?: H extends any[] ? T : never
+  ): T extends Record<string, infer V>
+    ? unknown extends V
+      ? Record<string, string | string[]>
+      : H extends any[]
+      ? Record<string, string | string[] | V>
+      : Record<string, string | string[]>
+    : never;
+}
```
Comment (tsctx) [inline types/util.d.ts:42]: It is a little more complicated, but it can handle all cases.

Comment (tsctx) [inline types/util.d.ts:42]: It is a little more complicated, but it can handle all cases.

**Review by ronag — APPROVED** [review]

**Comment by metcoder95** [comment]
@ronag @mcollina did we make changes recently to the BalancedPool?
I'm seeing other PRs starting to fail because of its tests

**Comment by tsctx** [comment]
@metcoder95 
This is due to node. This happens in the latest version (v18.19.0), but not in the previous version (v18.18.2).

**Pull request merged**